runtime/shared/shrclssup.c

```diff
@@ -141,6 +141,20 @@ IDATA J9VMDllMain(J9JavaVM* vm, IDATA stage, void* reserved)
 					runtimeFlags |= J9SHR_RUNTIMEFLAG_DISABLE_BCI;
 				}
 
+				/* Check for -XX:+PortableShareClasses and -XX:-PortableShareClasses; whichever comes later wins.
+				 * These options should be checked before parseArgs() to allow -Xshareclasses:portable to override this option.
+				 * 
+				 * Note: Please also change the function checkArgsConsumed() in runtime/vm/jvminit.c when adding new options,
+				 * in order to quietly consume the options if it is used without -Xshareclasses
+				 */
+				argIndex1 = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXSHARECLASSESPORTABLE, NULL);
+				argIndex2 = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXNOSHARECLASSESPORTABLE, NULL);
+				if (argIndex1 > argIndex2) {
+					vm->sharedCacheAPI->sharedCachePortable = TRUE;
+				} else if (argIndex2 > argIndex1) {
+					vm->sharedCacheAPI->sharedCachePortable = FALSE;
+				}
```

Comment:
what is the default initialization value of `vm->sharedCacheAPI->sharedCachePortable` if we don't execute either of these paths?

Reply:
Default initialization value is 0. 
Near the beginning of this function:
```cpp
memset(vm->sharedCacheAPI, 0, sizeof(J9SharedCacheAPI));
```